ledger/complete/wal/checkpointer.go

```diff
@@ -194,7 +194,13 @@ func (c *Checkpointer) Checkpoint(to int, targetWriter func() (io.WriteCloser, e
 	if err != nil {
 		return fmt.Errorf("cannot generate writer: %w", err)
 	}
-	defer writer.Close()
+	defer func() {
+		closeErr := writer.Close()
+		// Return close error if there isn't any prior error to return.
+		if err == nil {
+			err = closeErr
+		}
```

Comment:
[optional suggestion] In case `StoreCheckpoint` errored, we'll drop the `closeErr` here. You could use a `multierror` here to capture both:
```suggestion
		} else {
		    err = multierror.Append(err, closeErr)
		}
```

Reply:
Great suggestion!  I'll merge this as-is so it can be included in Canary.  I'll review which additional ledger functions might benefit from using `multierror` for a new issue/pr.